com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs

```diff
@@ -1467,67 +1448,76 @@ internal void HandleApproval(ulong ownerClientId, bool createPlayerObject, uint?
                     return;
                 }
 
-                // Inform old clients of the new player
-                foreach (KeyValuePair<ulong, NetworkClient> clientPair in ConnectedClients)
+                // Separating this into a contained function call for potential further future separation of when this notification is sent.
+                NotifyPlayerConnected(ownerClientId, playerPrefabHash ?? NetworkConfig.PlayerPrefab.GetComponent<NetworkObject>().GlobalObjectIdHash);
+            }
+            else
+            {
+                PendingClients.Remove(ownerClientId);
+                NetworkConfig.NetworkTransport.DisconnectRemoteClient(ownerClientId);
+            }
+        }
+
+        /// <summary>
+        /// Notifies all existing clients that a new player has joined
+        /// </summary>
+        /// <param name="clientId">new player client identifier</param>
+        /// <param name="playerPrefabHash">the prefab GlobalObjectIdHash value for this player</param>
+        internal void NotifyPlayerConnected(ulong clientId, uint playerPrefabHash)
```

Comment:
I don't think this is really `NotifyPlayerConnected`. I think this is more along the lines of `send create object command for player object`. What I mean is this just spawns a network objects on other clients it does not notify them about a new player connecting.

Reply:
Either of these names make more sense?

- SendNewPlayerNetworkObject
- SpawnPlayerNetworkObject

Reply:
@LukeStampfli 
I renamed this to:
`        internal void ApprovedPlayerSpawn(ulong clientId, uint playerPrefabHash)`

What do you think?

Reply:
Yes that's much better 👍